diff --git a/src/trio/_core/__init__.py b/src/trio/_core/__init__.py
index c53d9d59..25ceb583 100644
--- a/src/trio/_core/__init__.py
+++ b/src/trio/_core/__init__.py
@@ -87,7 +87,7 @@ if sys.platform == "win32" or (
     )
 # Kqueue imports
 if (
-    sys.platform != "linux" and sys.platform != "win32" and sys.platform != "android"
+    sys.platform != "linux" and sys.platform != "win32" and sys.platform != "android" and not sys.platform.startswith("qnx")
 ) or (not _t.TYPE_CHECKING and "sphinx.ext.autodoc" in sys.modules):
     from ._run import current_kqueue, monitor_kevent, wait_kevent
 
diff --git a/src/trio/_core/_run.py b/src/trio/_core/_run.py
index 9ecdada1..ea16d251 100644
--- a/src/trio/_core/_run.py
+++ b/src/trio/_core/_run.py
@@ -3119,6 +3119,7 @@ if sys.platform == "win32":
 elif (
     sys.platform == "linux"
     or sys.platform == "android"
+    or sys.platform.startswith("qnx")
     or (not TYPE_CHECKING and hasattr(select, "epoll"))
 ):
     from ._generated_io_epoll import *
